src/psrsearch.jl

```diff
@@ -38,26 +36,25 @@ julia> z_n_binned([10., 0., 0., 0., 0.], 2)
 40.0
 ```
 """
-function z_n_binned(profile::Array{T,1}, n::Integer)::T where {T<:AbstractFloat}
+function z_n_binned(profile::AbstractVector{T}, n::Integer)::T where {T<:AbstractFloat}
     total = sum(profile)
     N = length(profile)
-    phase = Array(range(0, stop = N - 1) / N) * 2 * pi
+    phase = range(0, stop = N - 1) / N * 2 * pi
 
-    if total == 0
-        return 0
+    if iszero(total)
+        return zero(T)
     end
 
-    z = 0
+    z = zero(T)
     for k in range(1, stop = n)
-        s = 0
-        for i in range(1, stop = N)
-            s += profile[i] * sin(k * phase[i])
+        s = zero(T)
+        c = zero(T)
+        for i in eachindex(profile)
+            sk, ck = sincos(k * phase[i])
```

Comment:
`sincos` is an efficient function which computes `sin` and `cos` for the same argument at the same time: the two functions partly share the same kernel, so you can save some time.

A small benchmark to show the improvement:
```julia
julia> using BenchmarkTools

julia> function test1!(v, phases)
           for idx in eachindex(v)
               s = sin(phases[idx])
               c = cos(phases[idx])
               v[idx] = s ^ 2 + c ^ 2
           end
           return v
       end
test1! (generic function with 1 method)

julia> function test2!(v, phases)
           for idx in eachindex(v)
               s, c = sincos(phases[idx])
               v[idx] = s ^ 2 + c ^ 2
           end
           return v
       end
test2! (generic function with 1 method)

julia> phases = collect(0:0.0001:2pi);

julia> v = similar(phases);

julia> @btime test1!($(v), $(phases));
  972.412 μs (0 allocations: 0 bytes)

julia> @btime test2!($(v), $(phases));
  702.692 μs (0 allocations: 0 bytes)
```